src/core/packages/http/router-server-internal/src/route.ts

```diff
@@ -205,10 +206,14 @@ function routeSchemasFromRouteConfig<P, Q, B>(
   }
 }
 
-function getPostValidateEventMetadata(request: AnyKibanaRequest, routeInfo: RouteInfo) {
+function getPostValidateEventMetadata(
+  request: AnyKibanaRequest,
+  routeInfo: RouteInfo
+): PostValidationMetadata {
   return {
     deprecated: routeInfo.deprecated,
     isInternalApiRequest: request.isInternalApiRequest,
     isPublicAccess: isPublicAccessApiRoute(routeInfo),
```

Comment:
```suggestion
    isPublicAccess: routeInfo.access === 'public'
```

On alternative to the proposed changes is to preserve the previous behaviour instead of the using/fixing `isPublicAccessApiRoute` here.